File: src/main/java/SignalServer.java
import users.UsersRepository;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.InputStream;
import java.net.DatagramPacket;
import java.net.DatagramSocket;
import java.security.Security;
import java.util.Properties;

public class SignalServer {

    public static void main(String[] args) throws Exception {
        InputStream inputStream = null;
        try {
            inputStream = new FileInputStream("signal.properties");
        } catch (FileNotFoundException e) {
            System.err.println("Configuration file not found!");
            System.exit(1);
        }
        Security.addProvider(new org.bouncycastle.jce.provider.BouncyCastleProvider());
        Properties properties = new Properties();
        properties.load(inputStream);
        String userid = properties.getProperty("userid");
        String proxyboxid = properties.getProperty("proxyboxid");
        int port = Integer.parseInt(properties.getProperty("port"));
        DatagramSocket s = new DatagramSocket(port);
        UsersRepository ur = new UsersRepository("users.json");
        signal.Authentication ss = new signal.Authentication();
        while (true) {
            byte[] buff = new byte[4096];
            DatagramPacket p = new DatagramPacket(buff, buff.length);
            s.receive(p);
            ss.processMessage(p);
        }
    }

}

File: src/main/kotlin/SignalServer.kt
import kotlinx.coroutines.launch
import kotlinx.coroutines.runBlocking
import java.io.FileInputStream
import java.io.FileNotFoundException
import java.net.DatagramPacket
import java.net.DatagramSocket
import java.net.ServerSocket
import java.net.Socket
import java.util.*
import kotlin.system.exitProcess

fun main() = runBlocking {
    lateinit var inputStream: FileInputStream
    try {
        inputStream = FileInputStream("signal.properties")
    } catch (e: FileNotFoundException) {
        println("Configuration file not found!")
        exitProcess(1)
    }


    val properties = Properties()
    properties.load(inputStream)
    val userid = properties.getProperty("userid")
    val proxyboxid = properties.getProperty("proxyboxid")
    val port = properties.getProperty("port").toInt()
    val s = DatagramSocket(port)
    val ss = SignalServer(userid, proxyboxid, port, s)
    while (true) {
        val buff = ByteArray(4096)
        val p = DatagramPacket(buff,buff.size)
        s.receive(p)
        launch {
            ss.processMessage(p)
        }
    }
}

class SignalServer(val userid: String, val proxyboxid: String, val port: Int, s: DatagramSocket) {

    fun processMessage(p: DatagramPacket){

    }

    private fun doAuthentication(s: Socket) {
        respondHello(s)

        respondAuthentication(s)

        respondPayment(s)
    }
    private fun respondHello(s: Socket) {
        //Todo
    }

    private fun respondAuthentication(s: Socket) {
        TODO("Not yet implemented")
    }

    private fun respondPayment(s: Socket) {
        TODO("Not yet implemented")
    }


}